tests/components/zwave_js/test_api.py

```diff
@@ -4987,3 +4989,39 @@ async def test_invoke_cc_api(
     msg = await ws_client.receive_json()
     assert not msg["success"]
     assert msg["error"] == {"code": "NotFoundError", "message": ""}
+
+
+async def test_get_integration_settings(
+    hass: HomeAssistant,
+    integration: MockConfigEntry,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test that the get_integration_settings WS API call works."""
+    ws_client = await hass_ws_client(hass)
+
+    # Test default settings
+    await ws_client.send_json_auto_id(
+        {
+            TYPE: "zwave_js/get_integration_settings",
+        }
+    )
+    msg = await ws_client.receive_json()
+    assert msg["success"]
+    assert msg["result"] == {CONF_INSTALLER_MODE: False}
+
+    # Test with installer_mode: true
+    # Unload the component
+    hass.config.components.remove(DOMAIN)
```

Comment:
I'd parametrize the test instead with `pytest.mark.parametrize` and set up the integration for both cases with `async_setup_component`. Ie don't use the `integration` fixture, but use the `client` fixture and add a `MockConfigEntry` before calling `async_setup_component`.